core/shared/src/main/scala/chartreuse/Plot.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2015 Creative Scala
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package chartreuse
+
+import chartreuse.layout.ScatterPlot
+import doodle.algebra.*
+import doodle.core.*
+import doodle.syntax.all.*
+
+/** A `Plot` is a collection of layers along with a title, legend, axes, and
+  * grid.
+  */
+final case class Plot[
+    A,
+    Alg <: Layout & Shape
+](
+    layers: List[Layer[A, Alg]],
+    plotTitle: String = "Plot Title",
+    xTitle: String = "X data",
+    yTitle: String = "Y data",
+    grid: Boolean = true
+) {
+  def addLayer(layer: Layer[A, Alg]): Plot[A, Alg] = {
+    copy(layers = layer :: layers)
+  }
+
+  def setPlotTitle(newPlotTitle: String): Plot[A, Alg] = {
```

Comment:
The convention in other parts of Chartreuse (and Doodle) is to use the prefix `with`, not `set`, for builder style methods. See Layer for examples.